contracts/interfaces/IProver.sol

```diff
@@ -29,16 +19,7 @@ interface IProver is ISemver {
 
     /**
      * @notice Gets the proof mechanism type used by this prover
-     * @return ProofType enum indicating the prover's mechanism
-     */
```

Comment:
Changing the return type from a type-safe enum to a string reduces type safety and could break dependent code. This also makes it harder for callers to validate the prover type programmatically.